design/data-flow.md

```diff
@@ -0,0 +1,68 @@
+                - Check timestamp
+                    - https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4040
+                - Reject block.nVersion < 4 blocks
+                - Don't accept any forks from the main chain prior to last checkpoint
+                    - We will probably get this 'more naturally' and don't need to explicitly check it like this
+        - Calls `CheckBlock` also, w/ ProofVerifier::Disabled == true
+            - https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4211
+        - Calls `ContextualCheckBlock`, defined at https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4065
+            - For each transaction:
+                - Calls `ContextualCheckTransaction`, defined at https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L760
+                    - For Zebra, we should only be doing Transaction v4 (Sapling+) checks, because checkpointing
+                        - If we reach a point where we are doing contextual transaction validation for the other types, abort
+                    - Check that all transactions are unexpired
+                    - A coinbase transaction cannot have output descriptions
+                    - All Sapling outputs in coinbase transactions MUST have valid note commitments when recovered using a 32-byte array of zeroes as the outgoing viewing key. https://zips.z.cash/zip-0213#specification
+                    - Compute SigHash on data to be signed and _previous_ data to be signed
+                    - Check joinsplit signature on data to be signed
+                        - If fail, check on previous data to be signed : https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L971
+                    - Check Spend/Output Descriptions 
+                        - Check Spends: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L998
+                        - Check Outputs: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L1018
+                        - Check final balances: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L1036
+                - Calls `IsFinalTx`, defined at https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L675
+                    - Calls tx::IsFinal(): https://github.com/zcash/zcash/blob/6d9573c66e33b5b742c49ab35e816d74d4ab55b1/src/primitives/transaction.h#L401
+            - Enforce BIP 34: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4090
+            - If blockheight < the block just before the first subsidy halving block, dispense Founder's Reward: https://github.com/zcash/zcash/blob/ab2b7c0969391d8a57d90d008665da02f3f618e7/src/main.cpp#L4104
```

Comment:
The coinbase, founders reward, and dev fund don't need any verification context, so I've opened ticket #801 for them. I also added the relevant consensus parameters to #565.